src/deepsparse/v2/text_generation/pipeline.py

```diff
@@ -109,17 +114,22 @@ def __init__(
             sequence_length=sequence_length,
         )
         compile_prompt_logits = CompilePromptLogits()
-        """
-        prep_for_single_engine = PrepareforSingleEngine(
-            prompt_sequence_length=prompt_sequence_length,
+        autoregressive_preprocess = AutoRegressiveOperatorPreprocess(
             sequence_length=sequence_length,
+            prompt_sequence_length=prompt_sequence_length,
         )
-        """
-        autoregressive_preprocess = AutoRegressiveOperatorPreprocess(
+        token_generater = TokenGeneratorOperator()
+        prep_for_generation = PrepareGeneration(
             sequence_length=sequence_length,
             prompt_sequence_length=prompt_sequence_length,
+            token_generator=token_generater,
```

Comment:
can we not just make the generator directly in this function?

Reply:
+1, I do not like the idea of creating a separate `TokenGeneratorOperator()`, let's keep `TokenGenerator` being a class, that is being instantiated inside the `PrepareGeneration` operator.